Content.Shared/Chat/ChatModifiers/EntityNameHeaderChatModifier.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Content.Shared.Chat.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Chat.ChatModifiers;
+
+/// <summary>
+/// Adds a [EntityNameHeader="name"] tag in front of the message.
+/// The string inside the tag is based upon the sending entity, and may be changed via certain transformation events (e.g. voicemasks).
+/// Once this tag is processed, it gets replaced with the string inside the tag.
+/// </summary>
+[Serializable]
+[DataDefinition]
+public sealed partial class EntityNameHeaderChatModifier : ChatModifier
+{
+    [Dependency] private readonly EntityManager _entityManager = default!;
+
+    public override FormattedMessage ProcessChatModifier(FormattedMessage message, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
+
+        if (channelParameters.TryGetValue(DefaultChannelParameters.SenderEntity, out var sender) && sender is EntityUid)
+        {
+            var senderEntity = (EntityUid)sender;
+            MetaDataComponent? metaData = null;
+            string? name;
```

Comment:
you are not using it in next block, don't declare it here then :D declare it later